sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -221,6 +222,19 @@ public PlannerResult plan() throws SqlParseException, ValidationException, RelCo
     final SqlNode validatedQueryNode = planner.validate(parameterizedQueryNode);
     final RelRoot rootQueryRel = planner.rel(validatedQueryNode);
 
+    if (parsed.getInsertNode() != null) {
+      // Check that there are no unnamed columns in the insert.
+      // We do this validation here since we need the rel nodes for this.
+      for (Pair<Integer, String> field : rootQueryRel.fields) {
+        if (UNNAMED_COLUMN_PATTERN.matcher(field.right).matches()) {
+          throw new ValidationException("Cannot ingest unnamed expressions that do not have an alias "
```

Comment:
nit: I think `Cannot ingest unnamed expressions that do not have an alias` can be converted to `Cannot ingest  expressions that do not have an alias`

Reply:
Changed